### PR TITLE
MTDSA 12268 -  Testable -API#1428 (#30) - Create a Historic Non-FHL UK Property Income & Expenses Period Summary Models

### PR DESCRIPTION
**Pull request by SamuelJacobsHMRC**

### CODE REVIEW

File: app/v2/models/request/common/ukNonFhlProperty/UkNonFhlPropertyIncome.scala [hunk 1]
```diff
@@ -38,6 +37,15 @@ object UkNonFhlPropertyIncome {
       (JsPath \ "taxDeducted").writeNullable[BigDecimal] and
       (JsPath \ "otherIncome").writeNullable[BigDecimal] and
       (JsPath \ "ukOtherRentARoom").writeNullable[UkPropertyIncomeRentARoom]
-  ) (unlift(UkNonFhlPropertyIncome.unapply))
+  )(unlift(UkNonFhlPropertyIncome.unapply))
+
+  val writesIFS: Writes[UkNonFhlPropertyIncome] = (
```
Comment (mattstephens-hmrc) [inline app/v2/models/request/common/ukNonFhlProperty/UkNonFhlPropertyIncome.scala:42]: Is this meant to have both writes and writesIFS?

Comment (mattstephens-hmrc) [reply app/v2/models/request/common/ukNonFhlProperty/UkNonFhlPropertyIncome.scala:42]: Can the // DONE comments be removed?

Comment (SamuelJacobsHMRC) [reply app/v2/models/request/common/ukNonFhlProperty/UkNonFhlPropertyIncome.scala:42]: > Is this meant to have both writes and writesIFS?

They do differ slightly - lines 48 and 62 are where the two writes differ. Open to a less verbose way of achieving this?

Comment (mattstephens-hmrc) [reply app/v2/models/request/common/ukNonFhlProperty/UkNonFhlPropertyIncome.scala:42]: In this case I think more verbose is going to be the less risky answer :disappointed:
(see Slack comments)

File: test/v2/models/request/createHistoricNonFhlUkPropertyPeriodSummary/CreateHistoricNonFhlUkPropertyPeriodSummaryRawDataSpec.scala [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+package v2.models.request.createHistoricNonFhlUkPropertyPeriodSummary
+
+import play.api.libs.json.Json
+import support.UnitSpec
+
+class CreateHistoricNonFhlUkPropertyPeriodSummaryRawDataSpec extends UnitSpec {
+
+  "writes" must {
+    "work" in {
+      Json.toJson(CreateHistoricNonFhlUkPropertyPeriodSummaryRawData(nino = "some-nino", body = Json.obj("bodyField" -> "value"))) shouldBe
```
Comment (mattstephens-hmrc) [inline test/v2/models/request/createHistoricNonFhlUkPropertyPeriodSummary/CreateHistoricNonFhlUkPropertyPeriodSummaryRawDataSpec.scala:26]: could this be split out a bit, so you've got
```
val expectedJson = ...
val result = ...
result shouldBe expectedJson
```

File: test/v2/models/request/createHistoricNonFhlUkPropertyPeriodSummary/CreateHistoricNonFhlUkPropertyPeriodSummaryRawDataSpec.scala [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+package v2.models.request.createHistoricNonFhlUkPropertyPeriodSummary
+
+import play.api.libs.json.Json
+import support.UnitSpec
+
+class CreateHistoricNonFhlUkPropertyPeriodSummaryRawDataSpec extends UnitSpec {
+
+  "writes" must {
+    "work" in {
```
Comment (mattstephens-hmrc) [inline test/v2/models/request/createHistoricNonFhlUkPropertyPeriodSummary/CreateHistoricNonFhlUkPropertyPeriodSummaryRawDataSpec.scala:25]: better if this is specific, e.g. writes must... write valid json

File: test/v2/models/request/createHistoricNonFhlUkPropertyPeriodSummary/UkNonFhlPropertyExpensesSpec.scala [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+  "reads" when {
+    "passed a valid JSON" should {
+      "return a valid model" in {
+        mtdJson.as[UkNonFhlPropertyExpenses] shouldBe requestBody
+      }
+    }
+  }
+  "writes" when {
+    "passed valid model" should {
```
Comment (mattstephens-hmrc) [inline test/v2/models/request/createHistoricNonFhlUkPropertyPeriodSummary/UkNonFhlPropertyExpensesSpec.scala:85]: could this be "when.. passed a valid object" or something? "Model" refers to the domain model, i.e. all the case classes and their respective rules & behaviour.

**Pull request merged**